File: yukon-server/LoadManagement/lmcicustomerbase.cpp
#include "precompiled.h"

#include "dbaccess.h"
#include "lmcicustomerbase.h"
#include "lmid.h"
#include "logger.h"
#include "loadmanager.h"
#include "resolvers.h"

using std::string;

extern ULONG _LM_DEBUG;

/*---------------------------------------------------------------------------
    Constructors
---------------------------------------------------------------------------*/
CtiLMCICustomerBase::CtiLMCICustomerBase() :
_customerid(0),
_customerdemandlevel(0),
_curtailamount(0),
_customerorder(0)
{
}

CtiLMCICustomerBase::CtiLMCICustomerBase(Cti::RowReader &rdr)
{
    restore(rdr);
}

CtiLMCICustomerBase::CtiLMCICustomerBase(const CtiLMCICustomerBase& customer)
{
    operator=(customer);
}

/*---------------------------------------------------------------------------
    Destructor
---------------------------------------------------------------------------*/
CtiLMCICustomerBase::~CtiLMCICustomerBase()
{
}

LONG CtiLMCICustomerBase::getCustomerId() const
{
    return _customerid;
}

const string& CtiLMCICustomerBase::getCompanyName() const
{
    return _companyname;
}

DOUBLE CtiLMCICustomerBase::getCustomerDemandLevel() const
{
    return _customerdemandlevel;
}

DOUBLE CtiLMCICustomerBase::getCurtailAmount() const
{
    return _curtailamount;
}

const string& CtiLMCICustomerBase::getCurtailmentAgreement() const
{
    return _curtailmentagreement;
}

const string& CtiLMCICustomerBase::getTimeZone() const
{
    return _time_zone;
}

LONG CtiLMCICustomerBase::getCustomerOrder() const
{
    return _customerorder;
}



CtiLMCICustomerBase& CtiLMCICustomerBase::setCustomerId(LONG id)
{
    _customerid = id;
    //do not notify observers of this!
    return *this;
}

CtiLMCICustomerBase& CtiLMCICustomerBase::setCompanyName(const string& name)
{
    _companyname = name;
    return *this;
}

CtiLMCICustomerBase& CtiLMCICustomerBase::setCustomerDemandLevel(DOUBLE cdl)
{
    _customerdemandlevel = cdl;
    return *this;
}

CtiLMCICustomerBase& CtiLMCICustomerBase::setCurtailAmount(DOUBLE amount)
{
    _curtailamount = amount;
    return *this;
}

CtiLMCICustomerBase& CtiLMCICustomerBase::setCurtailmentAgreement(const string& agreement)
{
    _curtailmentagreement = agreement;
    return *this;
}

CtiLMCICustomerBase& CtiLMCICustomerBase::setTimeZone(const string& timezone)
{

    _time_zone = timezone;
    return *this;
}

CtiLMCICustomerBase& CtiLMCICustomerBase::setCustomerOrder(LONG order)
{
    _customerorder = order;
    return *this;
}

/*---------------------------------------------------------------------------
    operator=
---------------------------------------------------------------------------*/
CtiLMCICustomerBase& CtiLMCICustomerBase::operator=(const CtiLMCICustomerBase& right)
{
    if( this != &right )
    {
        _customerid             = right._customerid;
        _companyname            = right._companyname;
        _customerdemandlevel    = right._customerdemandlevel;
        _curtailamount          = right._curtailamount;
        _curtailmentagreement   = right._curtailmentagreement;
        _time_zone              = right._time_zone;
        _customerorder          = right._customerorder;
    }

    return *this;
}

/*---------------------------------------------------------------------------
    operator==
---------------------------------------------------------------------------*/
int CtiLMCICustomerBase::operator==(const CtiLMCICustomerBase& right) const
{

    return( (getCustomerId() == right.getCustomerId()) && (getCustomerOrder() == right.getCustomerOrder()) );
}

/*---------------------------------------------------------------------------
    operator!=
---------------------------------------------------------------------------*/
int CtiLMCICustomerBase::operator!=(const CtiLMCICustomerBase& right) const
{

    return !(operator==(right));
}

/*---------------------------------------------------------------------------
    restore

    Restores given a Reader
---------------------------------------------------------------------------*/
void CtiLMCICustomerBase::restore(Cti::RowReader &rdr)
{
    string tempBoolString;

    rdr["customerid"] >> _customerid;
    rdr["companyname"] >> _companyname;
    rdr["customerdemandlevel"] >> _customerdemandlevel;
    rdr["curtailamount"] >> _curtailamount;
    rdr["curtailmentagreement"] >> _curtailmentagreement;
    rdr["timezone"] >> _time_zone;
    rdr["customerorder"] >> _customerorder;
}

// Static Members

// Possible acknowledge statuses

File: yukon-client/loadcontrol/src/com/cannontech/loadcontrol/data/LMCICustomerBase.java
package com.cannontech.loadcontrol.data;

/**
 * @author aaron
 */
public class LMCICustomerBase 
{
	private Long customerID;
	private String companyName;
	private Double customerDemandLevel;
	private Double curtailAmount;
	private String curtailmentAgreement;
	private String timeZone;
	private Long customerOrder;
	
	/**
	 * Returns the companyName.
	 * @return String
	 */
	public String getCompanyName() {
		return companyName;
	}

	/**
	 * Returns the curtailAmount.
	 * @return Double
	 */
	public Double getCurtailAmount() {
		return curtailAmount;
	}

	/**
	 * Returns the curtailmentAgreement.
	 * @return String
	 */
	public String getCurtailmentAgreement() {
		return curtailmentAgreement;
	}

	/**
	 * Returns the customerDemandLevel.
	 * @return Double
	 */
	public Double getCustomerDemandLevel() {
		return customerDemandLevel;
	}

	/**
	 * Returns the customerID.
	 * @return Long
	 */
	public Long getCustomerID() {
		return customerID;
	}

	/**
	 * Returns the customerOrder.
	 * @return Long
	 */
	public Long getCustomerOrder() {
		return customerOrder;
	}

	/**
	 * Returns the timeZone.
	 * @return String
	 */
	public String getTimeZone() {
		return timeZone;
	}

	/**
	 * Sets the companyName.
	 * @param companyName The companyName to set
	 */
	public void setCompanyName(String companyName) {
		this.companyName = companyName;
	}

	/**
	 * Sets the curtailAmount.
	 * @param curtailAmount The curtailAmount to set
	 */
	public void setCurtailAmount(Double curtailAmount) {
		this.curtailAmount = curtailAmount;
	}

	/**
	 * Sets the curtailmentAgreement.
	 * @param curtailmentAgreement The curtailmentAgreement to set
	 */
	public void setCurtailmentAgreement(String curtailmentAgreement) {
		this.curtailmentAgreement = curtailmentAgreement;
	}

	/**
	 * Sets the customerDemandLevel.
	 * @param customerDemandLevel The customerDemandLevel to set
	 */
	public void setCustomerDemandLevel(Double customerDemandLevel) {
		this.customerDemandLevel = customerDemandLevel;
	}

	/**
	 * Sets the customerID.
	 * @param customerID The customerID to set
	 */
	public void setCustomerID(Long customerID) {
		this.customerID = customerID;
	}

	/**
	 * Sets the customerOrder.
	 * @param customerOrder The customerOrder to set
	 */
	public void setCustomerOrder(Long customerOrder) {
		this.customerOrder = customerOrder;
	}

	/**
	 * Sets the timeZone.
	 * @param timeZone The timeZone to set
	 */
	public void setTimeZone(String timeZone) {
		this.timeZone = timeZone;
	}

	/**
	 * @see java.lang.Object#toString()
	 */
	public String toString() {
		return getCompanyName();
	}

	public boolean isRampingIn()
	{
		return false;
	}

	public boolean isRampingOut()
	{
		return false;
	}

}
